File: backend/ManageMe/Database/Models/Feature.cs
using ManageMe.Enums;

namespace ManageMe.Database.Models;

public class Feature
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = null!;
    public Priority Priority { get; set; }
    public State State { get; set; }
    public int UniqueEntityId { get; set; }
    public ICollection<Task> Tasks { get; set; } = new List<Task>();
}

File: manage-me/src/app/models/Feature.ts
import { Task } from "./Task";

export class Feature {
    id?: number;
    name: string;
    description: string;
    priority: Priority;
    state: State;
    tasks: Task[]

    constructor(name: string, description: string, priority: Priority, state: State, tasks: Task[] = [], id?: number) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.priority = priority;
        this.state = state;
        this.tasks = tasks
    }
}


export enum Priority {
    High = 0,
    Medium = 1,
    Low = 2
}

export enum State {
    Todo = 0,
    Doing = 1,
    Done = 2
}